src/classes/tmDaoProvisioning.cls

```diff
@@ -0,0 +1,174 @@
+                                                        WHERE sfbase__PaymentMethod__c = :tmConstants.CONTRACT_PAYMENT_METHOD
+                                                        AND sfbase__RevenueOwner__c = :revOwner
+                                                        LIMIT 1];
+            if (paymentMethodMapping.size() > 0) {
+                paymentTerm = paymentMethodMapping.get(0).sfbase__DefaultPaymentTerm__c;
+            }                                                   
+        }
+        return paymentTerm;
+    }  
+
+    public static Id createOrderOnNewContract(Contract c, Order ord, List<OrderItem> ordItems, Id priceBookId) {
+        return provisioningHelper.doCreateOrderOnNewContractWithoutSharing(c, ord, ordItems, priceBookId);
+    }
+
+    public static Id createOrderOnExistingContract(Id contractId, Order ord, List<OrderItem> ordItems) {
+        return provisioningHelper.doCreateOrderOnNewContractWithoutSharing(contractId, ord, ordItems);
+    }
+    public static id upgradeOrderOnExistingContract(Id contractId, Order ord, List<OrderItem> ordItems, List<OrderItem> oldOrdItems) {
+        provisioningHelper.updateOrderItems(oldOrdItems);
+        return provisioningHelper.doCreateOrderOnNewContractWithoutSharing(contractId, ord, ordItems);
+    }
+    private static tmDaoProvisioningHelper provisioningHelper {
+        get {
+            if (provisioningHelper == null) {
+                provisioningHelper = new tmDaoProvisioningHelper();
+            }
+            return provisioningHelper;
+        }
+        private set;
+    }
+
+    private without sharing class tmDaoProvisioningHelper {
+        public Id doCreateOrderOnNewContractWithoutSharing(Contract c, Order ord, List<OrderItem> ordItems, Id pbId) {
+            c.RecordTypeId = recordTypeIds.get('Contract');
+            insert c;
+
+            //Activate contract
+            c.Status = tmConstants.CONTRACT_STATUS_ACTIVATED;
+            update c;
+
+            Id orderId = doCreateOrderOnNewContractWithoutSharing(c.Id, ord, ordItems);
+            return orderId;
+        }
+
+        public Id doCreateOrderOnNewContractWithoutSharing(Id contractId, Order ord, List<OrderItem> ordItems) {
+            ord.contractId = contractId;
+            ord.RecordTypeId = recordTypeIds.get('Order');
+
+            //1.Insert order
+            insert ord;
+            Id orderId = ord.Id;
+
+            //2.Insert order items
+            for (OrderItem oi : ordItems) {
+                oi.OrderId = orderId;
+            }
+            insert ordItems;
+
+            //3.Activate order
+            ord.Status = tmConstants.ORDER_STATUS_ACTIVATED;
+            ord.sfbase__IsAutoActivated__c = true;
+            update ord;
+
+            return orderId;
+        }
+        
+        public void updateOrderItems(List<OrderItem> oldOrdItems) {
+            if(oldOrdItems.size() > 0)
+                update oldOrdItems;  
```

Comment:
This DML statement is not surrounded by any try-catch block. Please add a try-catch block to ensure exceptions are handled properly.